### PR TITLE
updates

### PR DESCRIPTION
**Pull request by dhrumilp15**
# What this PR does

Refactor to dataclasses.

- [x] I have tested this code
- [x] I have requested a review from dhrumilp15 or another dev.

# Are there related Issues?

Nope

### CODE REVIEW

File: python/bot_code/cog.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+        Args:
+            ctx: The originating context.
+            files: The files to send to the context.
+        """
+        name = self.bot.user.name
+        avatar_url = self.bot.user.display_avatar.url
+        view = FileView(files)
+        embed = FileEmbed(files, name=name, avatar_url=avatar_url)
+        message = f"Found {files.files[0].filename} {'and more...' if len(files.files) > 1 else ''}"
```
Comment (carissa-tang) [inline python/bot_code/cog.py:266]: utils !!!

File: python/bot_code/bot_commands.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        A list of filenames that were deleted
+    """
+    serv_id = interaction.channel.id
+    if interaction.guild is not None:
+        serv_id = interaction.guild.id
+    files = await search_client.search(serv_id, query=query)
+    deleted_files = []
+    for file in files.files:
+        try:
+            onii_chan = bot.get_channel(int(file.channel_id))
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/bot_commands.py:32]: no

File: python/bot_code/bot_commands.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            onii_chan = interaction.guild.text_channels
+
+    bot_user = None
+    if interaction.guild is not None:
+        bot_user = interaction.guild.me
+
+    files = await search_client.search(
+        interaction,
+        onii_chan,
+        bot_user=bot_user,
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/bot_commands.py:68]: bot user never none

File: python/bot_code/views/help_view.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                                                            "Use commands to search, delete or remove files.\n"
+                                                            "Specifying search options can refine your queries.")
+        super().add_field(name="Search",
+                          value="Use `/search` to search for files. Selecting no options retrieves all files.",
+                          inline=False)
+        super().add_field(name="Delete", value="Use `/delete` to specify files to delete", inline=False)
+        super().add_field(name="Search Options", value="Use these to refine your search queries!", inline=False)
+        for opt in search_opts:
+            super().add_field(name=opt, value=search_opts[opt])
+        super().set_footer(text=f"Delivered by {name}, a better file manager for discord.", icon_url=avatar_url)
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/help_view.py:25]: ?

File: python/bot.py [hunk 5]
```diff
@@ -28,11 +23,9 @@
 intents = discord.Intents.default()
 intents.message_content = True
 bot = commands.Bot(command_prefix='fs!', intents=intents)
-# slash = SlashCommand(bot, sync_commands=True)
-
-
 debug_guild = [discord.Object(id=int(CONFIG.GUILD_ID))] if getattr(CONFIG, "GUILD_ID", None) else []
 
+
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot.py:28]: two newlines are recommended

File: python/bot_code/views/file_dropdown.py [hunk 6]
```diff
@@ -1,17 +1,17 @@
-from typing import List, Dict
 import discord
+from ..search.search_models import SearchResults, SearchResult
+
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/file_dropdown.py:3]: two newlines recommended

File: python/bot_code/bot_commands.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/bot_commands.py:63]: bot user is None only for dm channels, which is expected behaviour.

File: python/bot_code/bot_commands.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        A list of dicts of viewable files.
+    """
+    onii_chan = [query.channel]
+    if query.channel is None:
+        onii_chan = interaction.channel
+        if interaction.guild is not None:
+            onii_chan = interaction.guild.text_channels
+
+    bot_user = None
+    if interaction.guild is not None:
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/bot_commands.py:56]: refactored for cleanliness

File: python/bot_code/search/search_models.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                filetype = self.content_type
+                if filetype is None:
+                    value = value[value.index('/') + 1:]
+                    filetype = self.filetype
+                if filetype == "jpeg" or filetype == "jpg":
+                    filetype = "jpg"
+                if value == "jpeg" or value == "jpg":
+                    value = "jpg"
+                if value == 'image' or value == 'audio':
+                    if value not in filetype:
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/search/search_models.py:76]: refactored

File: python/bot_code/database/mongo_client.py [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+    async def get_file(self, file_id: int) -> dict:
+        """
+        Get the file url of the file id.
+
+        Args:
+            file_id: The id of the file to retrieve.
+
+        Returns:
+            The file url as a str.
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/database/mongo_client.py]: gone

File: python/bot_code/cog.py (hunk 1 above)
Comment (dhrumilp15) [reply python/bot_code/cog.py]: refactored

File: python/bot_code/models/query.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+import discord
+
+
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/models/query.py]: resolved

File: python/bot_code/search/async_search_client.py [hunk 11]
```diff
@@ -21,6 +17,6 @@ async def remove_doc(self, *args, **kwargs):
         pass
 
     @abstractmethod
-    async def search(self, *args, **kwargs):
+    async def search(self, *args, **kwargs) -> SearchResults:
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/search/async_search_client.py]: gone

File: python/bot_code/search/search_models.py [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+                        return False
+                else:
+                    if value != filetype:
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/search/search_models.py]: yes

File: python/bot_code/views/help_view.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/help_view.py]: resolved

File: python/bot_code/database/models.py [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+    type: str
+    timestamp: str
+
+    @staticmethod
+    def from_discord_interaction(command_type, interaction, query) -> 'Command':
+        channel = interaction.channel
+
+        source = channel.id
+        if channel.guild is not None:
+            source = channel.guild.id
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/database/models.py]: resolved

File: python/bot_code/database/models.py [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+        Args:
+            message: The discord.Message that contains this attachment
+            file: The discord.Attachment to convert to a dict.
+
+        Returns:
+            A dict that contains metadata about the attachment.
+        """
+        if '.' in file.filename:
+            filetype = file.filename[file.filename.rindex('.') + 1:]
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/database/models.py]: the earlier check forces a . to be in the filename

File: python/bot_code/views/file_embed.py [hunk 15]
```diff
@@ -0,0 +1,28 @@
+import discord
+from .file_view import FileView
+from ..search.search_models import SearchResults
+
+
+class FileEmbed(discord.Embed):
+
+    def __init__(self, files: SearchResults, name: str, avatar_url: str):
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/file_embed.py]: snake_case, not camel case

File: python/bot_code/views/file_embed.py [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+        """
+        Build file search embed.
+
+        Returns:
+            The help embed
+        """
+        first_file = files.files[0]
+        super().__init__(
+            title=f"Found {len(files.files)} file{'s' if len(files.files) > 1 else ''}",
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/file_embed.py]: refactored

[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/views/file_embed.py]: I think > is clearer for plural

File: python/bot_code/search/search_models.py [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+                    return False
+            elif key == "author" or key == "channel":
+                if getattr(self, key + "_id") != value.id:
+                    return False
+            elif key == "filetype":
+                filetype = self.content_type
+                if filetype is None:
+                    value = value[value.index('/') + 1:]
+                    filetype = self.filetype
+                if filetype == "jpeg" or filetype == "jpg":
```
[the remark this answers is not in the document]
Comment (dhrumilp15) [reply python/bot_code/search/search_models.py]: accepted

**Pull request merged**